OsmAnd/src/net/osmand/plus/views/layers/MapSelectionHelper.java

```diff
@@ -745,11 +736,13 @@ public static Amenity findAmenityByOsmId(@NonNull OsmandApplication app, @NonNul
 		QuadRect rect = MapUtils.calculateLatLonBbox(latLon.getLatitude(), latLon.getLongitude(), AMENITY_SEARCH_RADIUS);
 		List<Amenity> amenities = app.getResourceManager().searchAmenities(ACCEPT_ALL_POI_TYPE_FILTER, rect, true);
 
-		return findAmenityByOsmId(amenities, osmId);
+		return findAmenityByOsmId(amenities, osmId, latLon);
 	}
 
 	@Nullable
-	public static Amenity findAmenityByOsmId(@NonNull List<Amenity> amenities, long id) {
+	public static Amenity findAmenityByOsmId(@NonNull List<Amenity> amenities, long id, LatLon point) {
```

Comment:
Here could be an issue that we need to sort amenity by distance, if there are duplicates